cmake/modules/boards.cmake

```diff
@@ -0,0 +1,149 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# Validate board and setup boards target.
+#
+# This CMake module will validate the BOARD argument as well as splitting the
+# BOARD argument into <BOARD> and <BOARD_REVISION>.
+#
+# If a board implementation is not found for the specified board an error will
+# be raised and list of valid boards will be printed.
+#
+# If user provided board is a board alias, the board will be adjusted to real
+# board name.
+#
+# If board name is deprecated, then board will be adjusted to new board name and
+# a deprecation warning will be printed to the user.
+#
+# Outcome:
+# The following variables will be defined when this CMake module completes:
+#
+# - BOARD:          Board, without revision field.
+# - BOARD_REVISION: Board revision
+# - BOARD_DIR:      Board directory with the implementation for selected board
+# - ARCH_DIR:       Arch dir for extracted from selected board
+# - BOARD_ROOT:     BOARD_ROOT with ZEPHYR_BASE appended
+#
+# The following targets will be defined when this CMake module completes:
+# - board  : when invoked a list of valid boards will be printed
+#
+# Variable dependencies:
+# - BOARD_ROOT: CMake list of board roots containing board implementations
+# - ARCH_ROOT:  CMake list of arch roots containing arch implementations
+#
```

Comment:
Are optional variable dependencies intentionally not listed?

Reply:
> Are optional variable dependencies intentionally not listed?

No, cause after addressing this comment: https://github.com/zephyrproject-rtos/zephyr/pull/41301#discussion_r774468820
even those two variables listed here are optional. (Update text to reflect that.)
`BOARD_ROOT` and `ARCH_ROOT` may be empty.

Reply:
I assume you refer to: https://github.com/zephyrproject-rtos/zephyr/blob/6873c4c361a87514d57a1eef29342ec026d4b3de/cmake/modules/boards.cmake#L65

I have update the description.